tsl/src/continuous_aggs/common.c

```diff
@@ -1332,55 +1336,27 @@ build_union_query(CAggTimebucketInfo *tbinfo, int matpartcolno, Query *q1, Query
 	/*
 	 * If there is join in CAgg definition then adjust varno
 	 * to get time column from the hypertable in the join.
-	 *
-	 * In case of joins it is enough to check if the first node is not RangeTblRef,
-	 * because the jointree has RangeTblRef as leaves and JoinExpr above them.
-	 * So if JoinExpr is present, it is the first node.
-	 * Other cases of join i.e. without explicit JOIN clause is confirmed
-	 * by reading the length of rtable.
 	 */
-	if (list_length(q2->rtable) == CONTINUOUS_AGG_MAX_JOIN_RELATIONS ||
-		!IsA(linitial(q2->jointree->fromlist), RangeTblRef))
-	{
-		Oid normal_table_id = InvalidOid;
-		RangeTblEntry *rte = NULL;
-		RangeTblEntry *rte_other = NULL;
+	varno = list_length(q2->rtable);
 
-		if (list_length(q2->rtable) == CONTINUOUS_AGG_MAX_JOIN_RELATIONS)
-		{
-			RangeTblRef *rtref = linitial_node(RangeTblRef, q2->jointree->fromlist);
-			rte = list_nth(q2->rtable, rtref->rtindex - 1);
-			RangeTblRef *rtref_other = lsecond_node(RangeTblRef, q2->jointree->fromlist);
-			rte_other = list_nth(q2->rtable, rtref_other->rtindex - 1);
-		}
-		else if (!IsA(linitial(q2->jointree->fromlist), RangeTblRef))
+	if (list_length(q2->rtable) > 1 || !IsA(linitial(q2->jointree->fromlist), RangeTblRef))
```

Comment:
When does it happen that the first jointree element is not a RangeTblRef (therefore a JoinExpr), but the range table contains one entry? Might be good to add an explanation to the comment.

Reply:
Ooops... it was a leftover from previous messy code. Make no sense now, removed!